packages/playwright-core/src/server/injected/clock.ts

```diff
@@ -119,9 +119,21 @@ export class ClockController {
 
   performanceNow(): DOMHighResTimeStamp {
     this._replayLogOnce();
+    this._syncRealTimeBusyLoop();
```

Comment:
Shouldn't we do the same for `Date.now()` and `new Date()` then?

Reply:
I personally would start minimal for now and add it to more stuff if it bubbles up. If you think the likelihood of a user doing this vs. doing the same with `Date.now()` or `new Date()` we should do it for them as well.

Reply:
i saw more busy loops on Date.now than performance.now